client.py

```diff
@@ -218,31 +229,53 @@ def exit():
     """
     
     global CONNECTED
+    requestNumber=0
     
     t = time.time()
-    while time.time() - t < DISCONNECTION_WAITING_TIME:
+    while time.time() - t < DISCONNECTION_WAITING_TIME and requestNumber<MAX_REQUESTS:
+        requestNumber+=1
         if send("DISCONNECTION " + USERNAME + " END") == "DISCONNECTED " + USERNAME + " END":
             break
     
+    if requestNumber>=MAX_REQUESTS:
+        exitError()
+    
     CONNECTED = False
 
+def exitError():
+    """Exit the game
+        Called if there has been a problem with the server"""
+        
+    global CONNECTED
+    
+    print("Sorry a problem occured...")
+    
+    CONNECTED=False
+
 
 
 def main():
     """Main function launching the parallel threads to play the game and communicate with the server.
     """
     
     global CONNECTED
+    requestNumber=0
     
-    while not CONNECTED:
+    
+    while not CONNECTED and requestNumber<MAX_REQUESTS:
         CONNECTED = connect()
         time.sleep(WAITING_TIME)
+        requestNumber+=1
```

Comment:
Shouldn't the `requestNumber` reset if the connection occurs correctly ?

Reply:
since it is only in this function it doesn't need to be reset, however I just saw that if the last request (MAX_REQUEST) works it still count as a failure so I should chage that

Reply:
Ok my bad everything's fine, I thought it was the same used later when trying to extract players from the message received from the server.